internal/redirecturi/patterns.go

```diff
@@ -0,0 +1,164 @@
+package redirecturi
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/gobwas/glob"
+)
+
+// RedirectPatternType represents the type of redirect pattern
+type RedirectPatternType int
+
+const (
+	// PatternTypeExact represents an exact URL match
+	PatternTypeExact RedirectPatternType = iota
+	// PatternTypeDomainOnly represents a domain-only pattern
+	PatternTypeDomainOnly
+	// PatternTypeSchemeOnly represents a scheme-only pattern
+	PatternTypeSchemeOnly
+	// PatternTypeWildcard represents a wildcard pattern
+	PatternTypeWildcard
+)
+
+// String returns the string representation of the RedirectPatternType
+func (t RedirectPatternType) String() string {
+	switch t {
+	case PatternTypeExact:
+		return "exact"
+	case PatternTypeDomainOnly:
+		return "domain-only"
+	case PatternTypeSchemeOnly:
+		return "scheme-only"
+	case PatternTypeWildcard:
+		return "wildcard"
+	default:
+		return fmt.Sprintf("unknown(%d)", t)
+	}
+}
+
+// RedirectPattern represents a parsed redirect pattern
+type RedirectPattern struct {
+	Original    string
+	Type        RedirectPatternType
+	GlobPattern glob.Glob
+}
+
+// RedirectConfig holds the configuration for redirect patterns
+type RedirectConfig struct {
+	Patterns []RedirectPattern
+}
+
+// NewRedirectConfig creates a new RedirectConfig
+func NewRedirectConfig(patterns []RedirectPattern) *RedirectConfig {
+	return &RedirectConfig{
+		Patterns: patterns,
+	}
+}
+
+// CategorizePattern determines the type of a redirect uri and returns a RedirectPattern
+func CategorizePattern(pattern string) (RedirectPattern, error) {
+	rp := RedirectPattern{Original: pattern}
+
+	pattern = strings.TrimSpace(pattern)
+	if pattern == "" {
+		return rp, errors.New("pattern cannot be empty")
```

Comment:
Nit -- ideally prefix errors with some easy way to identify them, usually the package name or something close by